In [1]:
import warnings
warnings.filterwarnings('ignore')
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import json
import re

In [10]:
path = 'yelp_data/health_business00.csv'

In [11]:
df = pd.read_csv(path)
df.drop(['Unnamed: 0'],axis=1,inplace=True)

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5656 entries, 0 to 5655
Data columns (total 5 columns):
business_id        5656 non-null object
categories         5656 non-null object
name               5656 non-null object
state              5656 non-null object
categories_list    5656 non-null object
dtypes: object(5)
memory usage: 221.0+ KB


In [13]:
df.head()

business_id                                         categories  \
0  y-4xTZNKVm8mAZpiXMS5ZA              urologists, doctors, health & medical   
1  STcSSjPOZ0FWtjvjqw99Dg            doctors, orthopedists, health & medical   
2  WuiRF916fJl6aG5wC8qRAg  doctors, eyebrow services, tattoo removal, bea...   
3  WnRtn80rdsWOa2nWZ6G1iA  obstetricians & gynecologists, health & medica...   
4  SaR_j1ev_d55IBzORph_lQ  laser eye surgery/lasik, optometrists, ophthal...   

                                  name state  \
0                     Lauren Byrne, MD    AZ   
1          Black Mountain Orthopaedics    NV   
2                       Advanced Brows    NV   
3  Central Phoenix Women's Health Care    AZ   
4                             LASIK MD    QC   

                                     categories_list  
0    ['urologists', ' doctors', ' health & medical']  
1  ['doctors', ' orthopedists', ' health & medical']  
2  ['doctors', ' eyebrow services', ' tattoo remo...  
3  ['obstetricians & gynecologists', ' health & m...  
4  ['laser eye surgery/lasik', ' optometrists', '...

In [14]:
df = df[['business_id','categories','name','state']]
df.head()

business_id                                         categories  \
0  y-4xTZNKVm8mAZpiXMS5ZA              urologists, doctors, health & medical   
1  STcSSjPOZ0FWtjvjqw99Dg            doctors, orthopedists, health & medical   
2  WuiRF916fJl6aG5wC8qRAg  doctors, eyebrow services, tattoo removal, bea...   
3  WnRtn80rdsWOa2nWZ6G1iA  obstetricians & gynecologists, health & medica...   
4  SaR_j1ev_d55IBzORph_lQ  laser eye surgery/lasik, optometrists, ophthal...   

                                  name state  
0                     Lauren Byrne, MD    AZ  
1          Black Mountain Orthopaedics    NV  
2                       Advanced Brows    NV  
3  Central Phoenix Women's Health Care    AZ  
4                             LASIK MD    QC

## Investigate df.state

- 

In [15]:
df.state.value_counts(dropna=False)

AZ     2818
NV     1581
NC      342
ON      332
PA      184
OH      166
AB       83
WI       73
QC       30
IL       21
SC       20
NI        3
CA        1
NYK       1
OR        1
Name: state, dtype: int64

# Filter only US location
### `sorry Canada`

Remove
- ON
- QC
- AB
- NI
- NYK

In [16]:
df.state.unique()

array(['AZ', 'NV', 'QC', 'ON', 'PA', 'NC', 'OH', 'IL', 'SC', 'WI', 'AB',
       'NI', 'CA', 'OR', 'NYK'], dtype=object)

In [17]:
on = df.state == 'ON'
qc = df.state == 'QC'
ab = df.state == 'AB'
ni = df.state == 'NI'
nyk = df.state == 'NYK'

In [18]:
df = df[~(on|qc|ab|ni|nyk)]
df = df.reset_index(drop=True)

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5207 entries, 0 to 5206
Data columns (total 4 columns):
business_id    5207 non-null object
categories     5207 non-null object
name           5207 non-null object
state          5207 non-null object
dtypes: object(4)
memory usage: 162.8+ KB


In [20]:
list(df.columns)

['business_id', 'categories', 'name', 'state']

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5207 entries, 0 to 5206
Data columns (total 4 columns):
business_id    5207 non-null object
categories     5207 non-null object
name           5207 non-null object
state          5207 non-null object
dtypes: object(4)
memory usage: 162.8+ KB


## Search our categories_list
- create df

```python
df.categories_list[0]
```

`"['urologists', ' doctors', ' health & medical']"`

In [22]:
df['categories_list'] = df.categories
df['categories_list']  = df['categories_list'].str.split(',')

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5207 entries, 0 to 5206
Data columns (total 5 columns):
business_id        5207 non-null object
categories         5207 non-null object
name               5207 non-null object
state              5207 non-null object
categories_list    5207 non-null object
dtypes: object(5)
memory usage: 203.5+ KB


In [25]:
df.head()

business_id                                         categories  \
0  y-4xTZNKVm8mAZpiXMS5ZA              urologists, doctors, health & medical   
1  STcSSjPOZ0FWtjvjqw99Dg            doctors, orthopedists, health & medical   
2  WuiRF916fJl6aG5wC8qRAg  doctors, eyebrow services, tattoo removal, bea...   
3  WnRtn80rdsWOa2nWZ6G1iA  obstetricians & gynecologists, health & medica...   
4  Q9v3zXGR2TduAKzn5B1BhQ  medical centers, skin care, health & medical, ...   

                                  name state  \
0                     Lauren Byrne, MD    AZ   
1          Black Mountain Orthopaedics    NV   
2                       Advanced Brows    NV   
3  Central Phoenix Women's Health Care    AZ   
4                Summerlin Dermatology    NV   

                                     categories_list  
0          [urologists,  doctors,  health & medical]  
1        [doctors,  orthopedists,  health & medical]  
2  [doctors,  eyebrow services,  tattoo removal, ...  
3  [obstetricians & gynecologists,  health & medi...  
4  [medical centers,  skin care,  health & medica...

In [34]:
%%time
# make sure that categories_list shows up as a list not as a string of lists
counts = dict()
for i in range(len(df.categories_list)):
    for j in df.categories_list[i]:
        j= j.lstrip()
        counts[j] = counts.get(j, 0) + 1
        
cat_reviews = pd.DataFrame.from_dict(counts,columns=['total_reviews'],orient='index')
cat_reviews.sort_values(by=['total_reviews'], ascending=False, inplace=True)

CPU times: user 234 ms, sys: 15.6 ms, total: 250 ms
Wall time: 258 ms


In [35]:
cat_reviews.info()

<class 'pandas.core.frame.DataFrame'>
Index: 322 entries, health & medical to discount store
Data columns (total 1 columns):
total_reviews    322 non-null int64
dtypes: int64(1)
memory usage: 5.0+ KB


In [36]:
cat_reviews

total_reviews
health & medical                           5207
doctors                                    5011
medical centers                             817
family practice                             632
beauty & spas                               594
urgent care                                 413
obstetricians & gynecologists               406
naturopathic/holistic                       382
medical spas                                362
internal medicine                           350
cosmetic surgeons                           349
sports medicine                             344
pediatricians                               333
ophthalmologists                            322
optometrists                                319
chiropractors                               290
shopping                                    290
physical therapy                            280
orthopedists                                263
skin care                                   262
dermatologists                              249
diagnostic services                         218
hospitals                                   194
hair removal                                193
pain management                             191
massage therapy                             190
weight loss centers                         188
eyewear & opticians                         187
podiatrists                                 156
acupuncture                                 155
...                                         ...
mediators                                     1
horseback riding                              1
wigs                                          1
otologists                                    1
kids activities                               1
water purification services                   1
building supplies                             1
heating & air conditioning/hvac               1
water heater installation/repair              1
plumbing                                      1
dialysis clinics                              1
churches                                      1
art classes                                   1
arts & crafts                                 1
video/film production                         1
taekwondo                                     1
american (traditional)                        1
hair extensions                               1
hotels                                        1
chimney sweeps                                1
air duct cleaning                             1
swimming pools                                1
massage schools                               1
pet hospice                                   1
landscaping                                   1
leisure centers                               1
soccer                                        1
web design                                    1
men's hair salons                             1
discount store                                1

[322 rows x 1 columns]

## Inspect cat_review dataframe
- with over 322 individual categories lets remove the unanted

In [37]:
list(cat_reviews.iloc[:].index)
remove_list = [
 'hair removal',
 'laser hair removal',

 'tattoo removal',
 'active life',
 'massage',


 'reiki',
 'cannabis clinics',
 'day spas',

 'traditional chinese medicine',
 'tattoo',

 'life coach',

 'permanent makeup',

 'trainers',
 'eyelash service',
 'local services',

 'yoga',

 'cosmetics & beauty supply',

 'pets',
 'cannabis dispensaries',


 'hair salons',
 'makeup artists',
 'food',
 'waxing',
 'veterinarians',

 'pet services',
 'alternative medicine',

 'gyms',

 'fashion',

 'eyebrow services',
 'piercing',
 'meditation centers',
 'medical cannabis referrals',

 'lawyers',
 'saunas',
 'supernatural readings',
 'health coach',
 'pilates',
 'shoe stores',
 'event planning & services',
 'sporting goods',
 'restaurants',
 'outdoor gear',
 'ayurveda',
 'contractors',
 'specialty food',
 'psychics',
 'jewelry',
 'pet stores',
 'advertising',
 'hotels & travel',
 'sports wear',
 'nail salons',
 'doulas',
 'herbal shops',
 'session photography',
 'photographers',
 'tai chi',
 'automotive',
 'emergency pet hospital',
 'sports clubs',
 'art galleries',
 'oxygen bars',
 'personal injury law',
 'spiritual shop',
 'health markets',
 'golf lessons',
 'float spa',
 'sunglasses',
 'martial arts',
 'transportation',
 'qi gong',
 'lice services',
 'lingerie',
 'rolfing',
 'photography stores & services',
 'grocery',
 'gift shops',
 'batting cages',
 'baby gear & furniture',
 'social clubs',
 'cannabis collective',
 'threading services',
 'tui na',
 'undersea/hyperbaric medicine',
 'juice bars & smoothies',
 'flowers & gifts',
 'business consulting',
 'divorce & family law',
 'home & garden',
 'preschools',
 'real estate services',
 'real estate agents',
 'self storage',
 'skilled nursing',
 'movers',
 'opera & ballet',
 'language schools',
 'windshield installation & repair',
 'career counseling',
 'tapas/small plates',
 'real estate',
 'team building activities',
 'bankruptcy law',
 'holistic animal care',
 'department stores',
 'mediterranean',
 'aerial fitness',
 'local flavor',
 'toxicologists',
 'employment agencies',
 'pet groomers',
 'pet boarding',
 'pet sitting',
 'climbing',
 'first aid classes',
 'fingerprinting',
 'child care & day care',
 'vocational & technical school',
 'sugaring',
 'shopping centers',
 'cosmetology schools',
 'officiants',
 'vegetarian',
 'nightlife',
 'bars',
 'diners',
 'shoe repair',
 'home decor',
 'farmers market',
 'driving schools',
 'taxis',
 'pizza',
 'furniture assembly',
 'security services',
 'towing',
 'furniture repair',
 'airport shuttles',
 'barbers',
 'convenience stores',
 'veterans organizations',
 'hair stylists',
 'general litigation',
 'feng shui',
 'personal chefs',
 'painters',
 'it services & computer repair',
 'telecommunications',
 'public transportation',
 'food delivery services',
 'travel services',
 'auto parts & supplies',
 'car stereo installation',
 'mediators',
 'horseback riding',
 'wigs',
 'kids activities',
 'water purification services',
 'building supplies',
 'heating & air conditioning/hvac',
 'water heater installation/repair',
 'plumbing',
 'art classes',
 'arts & crafts',
 'video/film production',
 'taekwondo',
 'american (traditional)',
 'hair extensions',
 'hotels',
 'chimney sweeps',
 'air duct cleaning',
 'swimming pools',
 'massage schools',
 'pet hospice',
 'landscaping',
 'leisure centers',
 'soccer',
 'web design',
 "men's hair salons",
 'discount store']

In [38]:
len(remove_list)

179

In [39]:
remove_health = df[df['categories'].apply(lambda x: pd.Series(x.split(', ')).isin(remove_list).any())]

In [40]:
remove_health.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 635 entries, 2 to 5185
Data columns (total 5 columns):
business_id        635 non-null object
categories         635 non-null object
name               635 non-null object
state              635 non-null object
categories_list    635 non-null object
dtypes: object(5)
memory usage: 29.8+ KB


In [41]:
remove_health

business_id  \
2     WuiRF916fJl6aG5wC8qRAg   
4     Q9v3zXGR2TduAKzn5B1BhQ   
21    2lAvN0x40Sgg5-rG8WEj3w   
23    si6nnecyffD85-m_oDolbw   
32    JqBjbYoWT_Q14qThOiKL4A   
38    1UCuElYKIgONqbe7FR9JVQ   
39    Gy5EvWbuRejXIg8B_4_S2w   
62    Icxhh649AppgoCWwIhmyKg   
67    Brk4gEtKuzGrkHYasRoyuw   
70    EoCKHx8RgrVrf_kieXi01g   
98    sAMQU5CtyGiG4ZwRczN61g   
121   4uHJ0IcUJZU4RkmOZSiXKw   
123   Hbx_ZmeWpBRQmXzwubhtqA   
143   QDG3eTp7CL6Rf3ySWym9vA   
158   Zp7ZkFZzkInYNAxna3hZ4g   
159   muriGdv1pnJaNZTQfZq9CQ   
161   eQZJj-gxNn9BBlVZaBAoLg   
162   aIxzy_P_5Ysz--BnRo0PPA   
175   EOzqxps3OpWmh6EYCHCl7w   
185   8u_PwSJyVO4uvrAE7VKH2A   
205   FNmTKUHrNa5ogARqboBxfw   
221   YeVQuSPaghZ4mDt9x0xhAQ   
225   mr5Ds305PzhExis8MZgziw   
229   lsBtqWLQizVJ3Gfxljyi5A   
231   Hy86VawhO-pXdLgJkrODKg   
237   PMOPoQZh8ANiqtdSXvlCzg   
238   fkeGoOYeVnnwZzVDby_TGg   
239   FGL-4F9q8f1su5gAt8g4CQ   
242   ChOsd2OKT68ZKXeLWED1aA   
257   05qUHkx1cOv9i8GFLhZqoQ   
...                      ...   
4940  8yiryCimz8XWcfFym61apQ   
4942  ap2TiuLkZnHDyslC4MVoqQ   
4944  tZSZTtFf6Auf3ijfx8KrzQ   
4948  m_AhrnuHJksoNIOI7sGV9g   
4965  z4vPdLY6L0iPsWy6SsXfZg   
4973  Vdb5D7PX3uFC3gjZcCyizQ   
4985  OL7Vy1Q59uLCwt_Gal2g9g   
4992  PrCIWEj4-88nMg2jSS4u4Q   
5005  oXa6V08grv-8swLkGUThKQ   
5015  PSvrN5IkDFem-1mkgYC0yw   
5023  zH9tS0W-lrye10798Wjvug   
5042  QKCoRVW-o0AOLZqUZq48Pw   
5050  5J4dmhNHhSKV-vCAYaydjA   
5060  O-WNye8_ZlgvI4HfIs5zZA   
5083  efI62THmRB2PvCyDF08BAQ   
5099  Vicyayl5aW6fC9oN7H4U8Q   
5106  mc3XvuqrcSnWueDItwkOZQ   
5109  OV69v-3LFg0M3VKPb8TRJQ   
5110  AB5e6BeYM8gAeOh9IJb0rQ   
5124  9853tSCOf4QCsYy4YEwjHA   
5133  WC9nPUvB3fkOFEbtNRzR-Q   
5136  qXywpRthTA6i3xZfi73x6Q   
5143  vMehSmLuDnf7jWS-zW-7vA   
5151  RCHPnip87i48Wpaixy1GCA   
5159  tOwA3qCTZ6Ke56hSTrOoqQ   
5162  -Nau05o3mhq_U4_VR21l7Q   
5176  u-wMJnxyIW4JWUua-0jfug   
5179  MAGtuWQ6FNZz3h7CBTQ-pA   
5184  KYvjCd1vcKpvngwWjby0kg   
5185  22Bbq9wo4kvUn-HE645zVg   

                                             categories  \
2     doctors, eyebrow services, tattoo removal, bea...   
4     medical centers, skin care, health & medical, ...   
21    dermatologists, shopping, doctors, skin care, ...   
23    colonics, doctors, saunas, health & medical, n...   
32    health & medical, acupuncture, yoga, active li...   
38    reiki, health & medical, physical therapy, doc...   
39    fitness & instruction, active life, profession...   
62    chiropractors, beauty & spas, massage, health ...   
67    doctors, active life, sports medicine, gyms, h...   
70    eyelash service, beauty & spas, skin care, hai...   
98    oxygen bars, iv hydration, beauty & spas, doct...   
121   dermatologists, health & medical, laser hair r...   
123   doctors, health & medical, naturopathic/holist...   
143   internal medicine, health & medical, beauty & ...   
158   tattoo, beauty & spas, eyelash service, doctor...   
159   beauty & spas, health & medical, tattoo remova...   
161   tai chi, massage therapy, traditional chinese ...   
162   tattoo removal, doctors, surgeons, health & me...   
175   physical therapy, sports medicine, doctors, ac...   
185   beauty & spas, doctors, health & medical, plas...   
205   health & medical, doctors, permanent makeup, b...   
221   health & medical, psychiatrists, counseling & ...   
225   sports medicine, beauty & spas, massage, healt...   
229   shopping, doctors, health & medical, shoe stor...   
231   health & medical, cannabis clinics, doctors, n...   
237   beauty & spas, doctors, tattoo removal, health...   
238   health & medical, doctors, advertising, weight...   
239   yoga, massage therapy, active life, hydrothera...   
242   naturopathic/holistic, professional services, ...   
257   doctors, permanent makeup, skin care, laser ha...   
...                                                 ...   
4940  doctors, health & medical, beauty & spas, natu...   
4942          tattoo removal, doctors, health & medical   
4944  naturopathic/holisti

In [42]:
remove_health.head()

business_id                                         categories  \
2   WuiRF916fJl6aG5wC8qRAg  doctors, eyebrow services, tattoo removal, bea...   
4   Q9v3zXGR2TduAKzn5B1BhQ  medical centers, skin care, health & medical, ...   
21  2lAvN0x40Sgg5-rG8WEj3w  dermatologists, shopping, doctors, skin care, ...   
23  si6nnecyffD85-m_oDolbw  colonics, doctors, saunas, health & medical, n...   
32  JqBjbYoWT_Q14qThOiKL4A  health & medical, acupuncture, yoga, active li...   

                         name state  \
2              Advanced Brows    NV   
4       Summerlin Dermatology    NV   
21  Linda Woodson Dermatology    NV   
23   Ancient Therapies Studio    AZ   
32             AZ Health Path    AZ   

                                      categories_list  
2   [doctors,  eyebrow services,  tattoo removal, ...  
4   [medical centers,  skin care,  health & medica...  
21  [dermatologists,  shopping,  doctors,  skin ca...  
23  [colonics,  doctors,  saunas,  health & medica...  
32  [health & medical,  acupuncture,  yoga,  activ...

In [43]:
remove_health.index

Int64Index([   2,    4,   21,   23,   32,   38,   39,   62,   67,   70,
            ...
            5133, 5136, 5143, 5151, 5159, 5162, 5176, 5179, 5184, 5185],
           dtype='int64', length=635)

In [44]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5207 entries, 0 to 5206
Data columns (total 5 columns):
business_id        5207 non-null object
categories         5207 non-null object
name               5207 non-null object
state              5207 non-null object
categories_list    5207 non-null object
dtypes: object(5)
memory usage: 203.5+ KB


In [45]:
df.head()

business_id                                         categories  \
0  y-4xTZNKVm8mAZpiXMS5ZA              urologists, doctors, health & medical   
1  STcSSjPOZ0FWtjvjqw99Dg            doctors, orthopedists, health & medical   
2  WuiRF916fJl6aG5wC8qRAg  doctors, eyebrow services, tattoo removal, bea...   
3  WnRtn80rdsWOa2nWZ6G1iA  obstetricians & gynecologists, health & medica...   
4  Q9v3zXGR2TduAKzn5B1BhQ  medical centers, skin care, health & medical, ...   

                                  name state  \
0                     Lauren Byrne, MD    AZ   
1          Black Mountain Orthopaedics    NV   
2                       Advanced Brows    NV   
3  Central Phoenix Women's Health Care    AZ   
4                Summerlin Dermatology    NV   

                                     categories_list  
0          [urologists,  doctors,  health & medical]  
1        [doctors,  orthopedists,  health & medical]  
2  [doctors,  eyebrow services,  tattoo removal, ...  
3  [obstetricians & gynecologists,  health & medi...  
4  [medical centers,  skin care,  health & medica...

## Remove unwanted rows from df by using the index of remove_health

In [46]:
df.drop(remove_health.index, inplace=True)

In [47]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4572 entries, 0 to 5206
Data columns (total 5 columns):
business_id        4572 non-null object
categories         4572 non-null object
name               4572 non-null object
state              4572 non-null object
categories_list    4572 non-null object
dtypes: object(5)
memory usage: 214.3+ KB


In [48]:
df.head()

business_id                                         categories  \
0  y-4xTZNKVm8mAZpiXMS5ZA              urologists, doctors, health & medical   
1  STcSSjPOZ0FWtjvjqw99Dg            doctors, orthopedists, health & medical   
3  WnRtn80rdsWOa2nWZ6G1iA  obstetricians & gynecologists, health & medica...   
5  EjypAFLWk7k5gk4ebPJANQ  psychologists, counseling & mental health, nat...   
6  yOJrDZU5zkxJoZOJTZXLvQ  neurologist, spine surgeons, cosmetic surgeons...   

                                  name state  \
0                     Lauren Byrne, MD    AZ   
1          Black Mountain Orthopaedics    NV   
3  Central Phoenix Women's Health Care    AZ   
5      West Valley Naturopathic Center    AZ   
6                    Michael Seiff, MD    NV   

                                     categories_list  
0          [urologists,  doctors,  health & medical]  
1        [doctors,  orthopedists,  health & medical]  
3  [obstetricians & gynecologists,  health & medi...  
5  [psychologists,  counseling & mental health,  ...  
6  [neurologist,  spine surgeons,  cosmetic surge...

In [49]:
df = df.reset_index(drop=True)
df.head()

business_id                                         categories  \
0  y-4xTZNKVm8mAZpiXMS5ZA              urologists, doctors, health & medical   
1  STcSSjPOZ0FWtjvjqw99Dg            doctors, orthopedists, health & medical   
2  WnRtn80rdsWOa2nWZ6G1iA  obstetricians & gynecologists, health & medica...   
3  EjypAFLWk7k5gk4ebPJANQ  psychologists, counseling & mental health, nat...   
4  yOJrDZU5zkxJoZOJTZXLvQ  neurologist, spine surgeons, cosmetic surgeons...   

                                  name state  \
0                     Lauren Byrne, MD    AZ   
1          Black Mountain Orthopaedics    NV   
2  Central Phoenix Women's Health Care    AZ   
3      West Valley Naturopathic Center    AZ   
4                    Michael Seiff, MD    NV   

                                     categories_list  
0          [urologists,  doctors,  health & medical]  
1        [doctors,  orthopedists,  health & medical]  
2  [obstetricians & gynecologists,  health & medi...  
3  [psychologists,  counseling & mental health,  ...  
4  [neurologist,  spine surgeons,  cosmetic surge...

In [50]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4572 entries, 0 to 4571
Data columns (total 5 columns):
business_id        4572 non-null object
categories         4572 non-null object
name               4572 non-null object
state              4572 non-null object
categories_list    4572 non-null object
dtypes: object(5)
memory usage: 178.7+ KB


In [51]:
df.categories_list[0]

['urologists', ' doctors', ' health & medical']

## Again

In [52]:
counts = dict()
for i in range(len(df.categories_list)):
    for j in df.categories_list[i]:
        j= j.lstrip()
        counts[j] = counts.get(j, 0) + 1

In [53]:
cat_reviews2 = pd.DataFrame.from_dict(counts,columns=['total_reviews'],orient='index')
cat_reviews2.sort_values(by=['total_reviews'], ascending=False, inplace=True)

In [55]:
cat_reviews2.info()

<class 'pandas.core.frame.DataFrame'>
Index: 136 entries, health & medical to sports psychologists
Data columns (total 1 columns):
total_reviews    136 non-null int64
dtypes: int64(1)
memory usage: 2.1+ KB


In [56]:
cat_reviews2

total_reviews
health & medical                          4572
doctors                                   4383
medical centers                            741
family practice                            600
urgent care                                406
obstetricians & gynecologists              390
internal medicine                          338
pediatricians                              318
ophthalmologists                           312
optometrists                               306
sports medicine                            286
orthopedists                               252
cosmetic surgeons                          251
chiropractors                              239
beauty & spas                              233
physical therapy                           221
shopping                                   211
diagnostic services                        208
naturopathic/holistic                      202
dermatologists                             188
hospitals                                  185
eyewear & opticians                        179
medical spas                               176
pain management                            156
podiatrists                                147
weight loss centers                        145
diagnostic imaging                         140
surgeons                                   138
dentists                                   110
skin care                                  107
...                                        ...
hepatologists                                4
speech therapists                            3
gerontologists                               3
behavior analysts                            3
body contouring                              3
personal care services                       2
medical transportation                       2
dietitians                                   2
educational services                         2
colleges & universities                      2
teeth whitening                              2
hospice                                      2
nephrologists                                2
parenting classes                            2
retirement homes                             2
phlebologists                                1
cpr classes                                  1
health insurance offices                     1
pharmacy                                     1
hearing aids                                 1
blood & plasma donation centers              1
hypnosis/hypnotherapy                        1
dialysis clinics                             1
churches                                     1
religious organizations                      1
sex therapists                               1
otologists                                   1
placenta encapsulations                      1
public services & government                 1
sports psychologists                         1

[136 rows x 1 columns]

In [57]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4572 entries, 0 to 4571
Data columns (total 5 columns):
business_id        4572 non-null object
categories         4572 non-null object
name               4572 non-null object
state              4572 non-null object
categories_list    4572 non-null object
dtypes: object(5)
memory usage: 178.7+ KB


# Review
- removed unwanted locations
- some unwanted locations still show up like naturopathic/holistic
- could be fix in future

## Final healthcare business
- 4570

## Save

In [60]:
df.to_csv('yelp_data/health_business01.csv')